utils/shared.js

```diff
@@ -1156,6 +1156,135 @@ const buildStreckPlaceholderData = (collectionId, streckTubeData) => {
     console.error(`Issue found in updateSpecimen() (ConnectFaas): Streck Tube not found in biospecimenData for collection Id ${collectionId}. Building placeholder data.`);
 }
 
+const validIso8601Format = /\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}\.\d{3}Z/;
+const validEmailFormat = /^[a-zA-Z0-9.!#$%&'*+"\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*\.[a-zA-Z]{2,63}$/;
+const validPhoneFormat = /^\d{10}$/;
+
+const queryListFields = {
+    firstName: [fieldMapping.firstName, fieldMapping.preferredName],
+    lastName: [fieldMapping.lastName],
+    allPhoneNo: [fieldMapping.homePhone, fieldMapping.cellPhone, fieldMapping.otherPhone],
+    allEmails: [fieldMapping.prefEmail, fieldMapping.additionalEmail1, fieldMapping.additionalEmail2],
+}
+
+const userProfileHistoryKeys = [
+    fieldMapping.firstName,
+    fieldMapping.middleName,
+    fieldMapping.lastName,
+    fieldMapping.suffix,
+    fieldMapping.preferredName,
+    fieldMapping.cellPhone,
+    fieldMapping.canWeVoicemailMobile,
+    fieldMapping.canWeText,
+    fieldMapping.homePhone,
+    fieldMapping.canWeVoicemailHome,
+    fieldMapping.otherPhone,
+    fieldMapping.canWeVoicemailOther,
+    fieldMapping.address1,
+    fieldMapping.address2,
+    fieldMapping.city,
+    fieldMapping.state,
+    fieldMapping.zip,
+  ];
+
+
+/**
+ * Check whether dataObj contains any of the fields in participantQueryListFields. If yes, handle the query list fields in the participant profile.
+ * @param {object} dataObj - the update object for the participant profile.
+ * @returns - true if any of the fields in participantQueryListFields are in dataObj. false otherwise.
+ */
+const checkForQueryFields = (dataObj) => {
+    for (const key in queryListFields) {
+        // Check if any field in the array for this key exists in dataObj
+        const hasQueryField = queryListFields[key].some(field => dataObj[field] !== undefined);
+        if (hasQueryField) {
+            return true;
+        }
+    }
+    return false; 
+}
+
+const updateQueryListFields = (dataObj, existingDocData) => {
+    let updatedQueryObj = existingDocData['query'] ? {...existingDocData['query']} : {};
+
+    for (const queryField in queryListFields) {
+        const isFieldPresentInDataObj = queryListFields[queryField].some(field => dataObj[field]);
+        
+        if (isFieldPresentInDataObj) {
+            let currentQueryArray = updatedQueryObj?.[queryField] ? [...updatedQueryObj[queryField]] : [];
+            
+            queryListFields[queryField].forEach(field => {
+                if (dataObj[field]) {
+                    currentQueryArray = updateQueryArray(dataObj[field], existingDocData?.[field], currentQueryArray);
+                }
+
+                if (currentQueryArray.length > 0) {
+                    updatedQueryObj[queryField] = currentQueryArray;
+                }
```

Comment:
Should this block of code be outside of `forEach` loop?

Reply:
Good call. Same result, better efficiency. Updated.